resolver/error_wrapper.go

```diff
@@ -0,0 +1,68 @@
+package resolver
+
+import (
+	"context"
+	"errors"
+	"net"
+)
+
+type errorWrapper struct {
+	resolver Resolver
+}
+
+func NewErrorWrapper(resolver Resolver) Resolver {
+	return &errorWrapper{resolver: resolver}
+}
+
+func (e *errorWrapper) LookupHost(ctx context.Context, host string) (addrs []string, err error) {
+	addrs, err = e.resolver.LookupHost(ctx, host)
+	return addrs, convertError(err)
+}
+
+func (e *errorWrapper) LookupIPAddr(ctx context.Context, host string) (records []net.IPAddr, err error) {
+	records, err = e.resolver.LookupIPAddr(ctx, host)
+	return records, convertError(err)
+}
+
+func (e *errorWrapper) LookupPort(ctx context.Context, network, service string) (port int, err error) {
+	port, err = e.resolver.LookupPort(ctx, network, service)
+	return port, convertError(err)
+}
+
+func (e *errorWrapper) LookupCNAME(ctx context.Context, host string) (cname string, err error) {
+	cname, err = e.resolver.LookupCNAME(ctx, host)
+	return cname, convertError(err)
+}
+
+func (e *errorWrapper) LookupSRV(ctx context.Context, service, proto, name string) (cname string, records []*net.SRV, err error) {
+	cname, records, err = e.resolver.LookupSRV(ctx, service, proto, name)
+	return cname, records, convertError(err)
+}
+
+func (e *errorWrapper) LookupMX(ctx context.Context, name string) (records []*net.MX, err error) {
+	records, err = e.resolver.LookupMX(ctx, name)
+	return records, convertError(err)
+}
+
+func (e *errorWrapper) LookupNS(ctx context.Context, name string) (records []*net.NS, err error) {
+	records, err = e.resolver.LookupNS(ctx, name)
+	return records, convertError(err)
+}
+
+func (e *errorWrapper) LookupTXT(ctx context.Context, name string) (records []string, err error) {
+	records, err = e.resolver.LookupTXT(ctx, name)
+	return records, convertError(err)
+}
+
+func (e *errorWrapper) LookupAddr(ctx context.Context, addr string) (names []string, err error) {
+	names, err = e.resolver.LookupAddr(ctx, addr)
+	return names, convertError(err)
+}
+
+func convertError(err error) error {
+	var dnsError *net.DNSError
+	if errors.As(err, &dnsError) && dnsError.IsNotFound {
+		return ErrNotFound
+	}
+	return err
+}
```

Comment:
I think all that could just be provided by a simple `resolver.IsNotFound(err error) bool`